prov/shm/src/smr_init.c

```diff
@@ -72,6 +72,40 @@ static void smr_resolve_addr(const char *node, const char *service,
 	(*addr)[*addrlen - 1]  = '\0';
 }
 
+/*
+ * The smr_shm_space_check is to check if there's enough shm space we
+ * need under /dev/shm.
+ * Here we use #core instead of SMR_MAX_PEERS, as it is the most likely
+ * value and has less possibility of failing fi_getinfo calls that are
+ * currently passing, and breaking currently working app
+ */
+static int smr_shm_space_check(size_t tx_count, size_t rx_count)
+{
+	struct statvfs stat;
+	char shm_fs[] = "/dev/shm";
+	uint64_t available_size, shm_size_needed;
+	int num_of_core, err;
+
+	num_of_core = get_nprocs_conf();
```

Comment:
We can use ofi_sysconf(_SC_NPROCESSORS_ONLN) here, which provides some portability.  I don't think we need to worry about possibly offline processors.

Reply:
Sure, done.